File: sources/Example_task_dep.11.f90
! @@name:	task_dep.11f90
! @@type:	F-free
! @@compilable:	yes, omp_5.0
! @@linkable:	no
! @@expect:	success

subroutine set_an_element(e, val)
    implicit none
    integer :: e, val

    e = val

end subroutine 

subroutine print_all_elements(v, n)
    implicit none
    integer :: n, v(n)

    print *, v

end subroutine 

subroutine parallel_computation(n)
    implicit none
    integer :: n
    integer :: i, v(n)
  
    !$omp parallel
    !$omp single
        do i=1, n
            !$omp task depend(out: v(i))
                 all set_an_element(v(i), i)
            !$omp end task
        enddo

        !$omp task depend(iterator(it = 1:n), in: v(it))
       !!$omp task depend(in: v(1:n)) Violates Array section restriction.
            call print_all_elements(v, n)
        !$omp end task

    !$omp end single
    !$omp end parallel
end subroutine

File: sources/Example_task_dep.11.c
/*
* @@name:       task_dep.11c
* @@type:       C
* @@compilable: yes
* @@linkable:   no
* @@expect:     success
* @@version: omp_5.0
*/
#include<stdio.h>

void set_an_element(int *p, int val) {
    *p = val;
}

void print_all_elements(int *v, int n) {
    int i;
    for (i = 0; i < n; ++i) {
        printf("%d, ", v[i]);
    }
    printf("\n");
}

void parallel_computation(int n) {
    int v[n];
    #pragma omp parallel
    #pragma omp single
    {
        int i;
        for (i = 0; i < n; ++i)
            #pragma omp task depend(out: v[i])
            set_an_element(&v[i], i);

        #pragma omp task depend(iterator(it = 0:n), in: v[it])
     // #pragma omp task depend(in: v[0:n]) Violates Array section restriction.
        print_all_elements(v, n);

    }
}

